### PR TITLE
Extend automated test matrix and rename lint

### LINKED ISSUE #78: [Bug] Installation on Mac
# 🐛 Bug

## To reproduce

1. pip install pykale[extras]
2. pip install .[extras] 
3. pip install -e .[dev]

** Stack trace/error message **
```
zsh: no matches found: .[extras]
zsh: no matches found: pykale[extras]
zsh: no matches found: .[dev]
```

## Environment
active environment : base
    active env location : /opt/anaconda3
            shell level : 2
       user config file : /Users/liudaotan/.condarc
 populated config files : /Users/liudaotan/.condarc
          conda version : 4.9.2
    conda-build version : 3.20.5
         python version : 3.8.5.final.0
       virtual packages : __osx=10.16=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /opt/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/anaconda3/pkgs
                          /Users/liudaotan/.conda/pkgs
       envs directories : /opt/anaconda3/envs
                          /Users/liudaotan/.conda/envs
               platform : osx-64
             user-agent : conda/4.9.2 requests/2.24.0 CPython/3.8.5 Darwin/20.3.0 OSX/10.16
                UID:GID : 501:20
             netrc file : None
           offline mode : False

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.7
 - OS (e.g., Linux): Mac (M1)
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): 
 - Python version: 3.8
 - CUDA/cuDNN version: null
 - GPU models and configuration: 
 - Any other relevant information:

## Additional context
When I change the code as following , every thing goes well.
1. pip install 'pykale[extras]'
2. pip install '.[extras]' 
3. pip install '-e .[dev]'

### PR DESCRIPTION
**Pull request by bobturneruk**
Fixes #78.

### Description
- Adds automated tests on Windows and OSX.
- Renames check to lint

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Comment by bobturneruk** [comment]
Python 3.9 apparently needs conda to install pytorch at present, so I've left it out of the test matrix.

**Comment by bobturneruk** [comment]
There's a couple of pending actions, but I'm sure they'll pass eventually!

**Comment by haipinglu** [comment]
@bobturneruk The pending checks may be due to some bug in the yml file that needs to be fixed. It is not running at all but showing as **Expected** so it seems that the checks will never complete.

**Comment by bobturneruk** [comment]
Will need to pass before merge. I'll check back on this.

**Comment by bobturneruk** [comment]
Might need a change in settings to drop old and require new.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -5,35 +5,41 @@ name: test
...
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    name: Run tests (${{ matrix.os }}, pytorch version ${{ matrix.pytorch-version }}, python version ${{ matrix.python-version }})
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
         pytorch-version: ["master", "stable"]
```
Comment (haipinglu) [inline .github/workflows/test.yml:19]: I think we don't need to test two versions. I never use the "master" version.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -5,35 +5,41 @@ name: test
...
-        bash <(curl -s https://codecov.io/bash)
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          if [[ ${{ matrix.pytorch-version }} = "master" ]]; then
+            pip install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html;
```
Comment (haipinglu) [inline .github/workflows/test.yml:31]: We can remove this option. The stable version is what we use.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -5,35 +5,41 @@ name: test
...
-    branches: [ master ]
+    branches: [master]
   pull_request:
-    branches: [ master ]
+    branches: [master]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    name: Run tests (${{ matrix.os }}, pytorch version ${{ matrix.pytorch-version }}, python version ${{ matrix.python-version }})
```
Comment (haipinglu) [inline .github/workflows/test.yml:14]: Remove "pytorch version"

File: README.md [hunk 4]
```diff
@@ -32,7 +32,7 @@
 
 ## Installation
 
-You should [install PyTorch](https://pytorch.org/get-started/locally/) matching your hardware first. To work on graphs, install [PyTorch Geometric](https://github.com/rusty1s/pytorch_geometric) first follow its official instructions.
+Pykale requires a Python version between 3.6 and 3.8. You should [install PyTorch](https://pytorch.org/get-started/locally/) matching your hardware first. To work on graphs, install [PyTorch Geometric](https://github.com/rusty1s/pytorch_geometric) first follow its official instructions.
```
Comment (haipinglu) [inline README.md:35]: PyKale (to be consistent)

Comment (bobturneruk) [reply README.md]: Thanks!

File: .github/workflows/test.yml (hunk 3 above)
Comment (bobturneruk) [reply .github/workflows/test.yml]: Done, thanks.

File: .github/workflows/test.yml (hunk 2 above)
Comment (bobturneruk) [reply .github/workflows/test.yml]: Done.

File: .github/workflows/test.yml (hunk 1 above)
Comment (bobturneruk) [reply .github/workflows/test.yml]: Done.

**Comment by bobturneruk** [comment]
Thanks for the review @haipinglu - can you have a check in the settings and see if there is an option to require the new checks and not the older ones, please?

**Review by haipinglu — APPROVED** [review]
Thanks. Now all the 11 checks are required:  3 os x 3 python + docs + lint (renamed). Let's see how it goes. If unnecessary, we could have less os x python version as required.

**Pull request merged**